plugins/events/rest.go

```diff
@@ -122,3 +125,72 @@ func eventsListHandler(w http.ResponseWriter, r *http.Request) {
 
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, events)
 }
+
+func checkConnection(c transaction.TxnCtx) error {
+	var req eventsapi.Webhook
+
+	if err := c.Get("req", &req); err != nil {
+		return err
+	}
+	e := gd2events.New("TEST", map[string]string{}, false)
+	return gd2events.WebhookPublish(&req, e)
+}
+
+func registerWebhookTestStepFuncs() {
+	var sfs = []struct {
+		name string
+		sf   transaction.StepFunc
+	}{
+		{"webhook-test.checkConnection", checkConnection},
+	}
+	for _, sf := range sfs {
+		transaction.RegisterStepFunc(sf.sf, sf.name)
+	}
+}
+
+func webhookTestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	var req eventsapi.Webhook
+
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(
+			ctx, w, http.StatusBadRequest,
+			errors.ErrJSONParsingFailed)
+		return
+	}
+
+	if req.URL == "" {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "webhook URL is required field")
+		return
+	}
+	allNodes, err := peer.GetPeerIDs()
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	txn := transaction.NewTxn(ctx)
+	defer txn.Done()
+
+	txn.Steps = []*transaction.Step{
+		{
+			DoFunc: "webhook-test.checkConnection",
+			Nodes:  allNodes,
+		},
+	}
+
+	if err := txn.Ctx.Set("req", &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	if err := txn.Do(); err != nil {
+		logger.WithError(err).Error("webhook-test transaction failed")
+		status, err := restutils.ErrToStatusCode(err)
```

Comment:
This can be used at other places as well right?, where ever the status is generic

Reply:
This function can't be used for system generated error msgs like failed unmarshal.